server/build/webpack.js

```diff
@@ -10,6 +10,12 @@ import WatchRemoveEventPlugin from './plugins/watch-remove-event-plugin'
 import DynamicEntryPlugin from './plugins/dynamic-entry-plugin'
 import DetachPlugin from './plugins/detach-plugin'
 
+// get all the environment variables
+const env = Object.keys(process.env).reduce(function (o, k) {
+  o['process.env.' + k] = JSON.stringify(process.env[k])
```

Comment:
Now user must use `process.env.NEXT_FOO` in order to use it.
But this doesn't work.

```js
const { NEXT_FOO} = process.env
```

CRA also have a fix for that. Try to use something like that.